common/beerocks/bcl/source/network/get_iface_stats.cpp

```diff
@@ -0,0 +1,243 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <bcl/network/network_utils.h>
+
+#include <easylogging++.h>
+
+#include <errno.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <net/if.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define IFLIST_REPLY_BUFFER 8192
+
+namespace beerocks {
+namespace net {
+
+/**
+ * @brief Netlink request type.
+ */
+struct nl_req_t {
+    /**
+     * Netlink message
+     */
+    struct nlmsghdr hdr;
+
+    /**
+     * "general form of address family dependent" message, i.e. how to tell which AF we are
+     * interested in. */
+    struct rtgenmsg gen;
+};
+
+/**
+ * @brief Gets interface statistics for the given network interface.
+ *
+ * Gets interface stats for given network interface from the specified Netlink message of type
+ * RTM_NEWLINK.
+ *
+ * @param[in] h Pointer to the Netlink message containing the data.
+ * @param[in] iface_name Name of the network interface.
+ * @param[in, out] iface_stats Interface statistics structure with values read.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool get_iface_stats(const struct nlmsghdr *h, const std::string &iface_name,
+                            sInterfaceStats &iface_stats)
+{
+    bool result = false;
+
+    struct ifinfomsg *iface = static_cast<ifinfomsg *>(NLMSG_DATA(h));
+
+    size_t length = 0;
+    if (h->nlmsg_len > NLMSG_LENGTH(sizeof(*iface))) {
+        length = h->nlmsg_len - NLMSG_LENGTH(sizeof(*iface));
+    }
+
+    /**
+     * Loop over all attributes of the RTM_NEWLINK message
+     */
+    for (const struct rtattr *attribute = IFLA_RTA(iface); RTA_OK(attribute, length);
+         attribute                      = RTA_NEXT(attribute, length)) {
+        switch (attribute->rta_type) {
+        case IFLA_IFNAME:
+            /**
+             * This message contains the stats for the interface we are interested in
+             */
+            if (0 == std::strncmp(iface_name.c_str(), static_cast<char *>(RTA_DATA(attribute)),
+                                  iface_name.length() + 1)) {
+                result = true;
+            }
+            break;
+        case IFLA_STATS:
```

Comment:
Not sure if possible but consider those scenarios:  
1. You receive `IFLA_IFNAME`, then the next attribute is bad -> you return true even though you failed.
2. You receive `IFLA_IFNAME`, `IFLA_STATS` and then another set of those for different `iface_name` -> Since you already set the `result = true`. you will run the `iface_stats` over with irrelevant info.

Both of those can be fixed by:
line 96: `return true;`
line 102: `return false;`

Reply:
Done